### ISSUE TITLE
Update the types.sexpr module for the fixed JBUGs when moving to 2201.3.0

### ISSUE THREAD

**Issue by gayaldassanayake** [ballerina-platform/nballerina#1093]
**Description:**
The testerina framework was re-written in ballerina. When the nballerina tests were run with the re-written testerina framwork, the following error occured during the execution of `nballerina.types.sexpr:typeTests`. This is a data driven test and the error occurs during the passing of the parameters to the test from the data provider.

```
[fail data provider for the function typeTests]
                        incompatible types: '["cell",wso2/nballerina.types.sexpr:0:Type,CellMutability]' cannot be cast to '["mapping",[nballerina.comm.sexpr:Str,wso2/nballerina.types.sexpr:0:Type][],wso2/nballerina.types.sexpr:0:Type]'
```

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**
- Build the `testerina-migration` branch of the ballerina-lang repo
- Use that build to run the tests of the `types.sexpr` module of the `nballerina` package with the following command
``` bash
/path/to/build/ballerina-lang/distribution/zip/jballerina-tools/build/extracted-distributions/jballerina-tools-2201.3.0-SNAPSHOT/bin/bal test --tests "nballerina.types.sexpr:typeTests"
```

**Related Issues:**

**Comment by lochana-chathura** [comment]
This turned out not to be due to the new testerina rewrite.
Causing due to the new behavior of match pattern with the fixes introduced in the PR ballerina-platform/ballerina-lang#37235.

Need to update the following places with the update 3. 

https://github.com/ballerina-platform/nballerina/blob/066686b98e29d4fb507fdd9a3e8db03f3d924141/compiler/modules/types.sexpr/convert.bal#L71-L74
https://github.com/ballerina-platform/nballerina/blob/066686b98e29d4fb507fdd9a3e8db03f3d924141/compiler/modules/types.sexpr/convert.bal#L94-L95

**Comment by gayaldassanayake** [comment]
Fixed for the testerina-migration branch with https://github.com/ballerina-platform/ballerina-lang/pull/37870

**Comment by manuranga** [comment]
Will be done as part of #1132

**Comment by heshanpadmasiri** [comment]
Fixed with #1134
